examples/esp-chat-react/js/components/ChatApp.react.js

```diff
@@ -0,0 +1,39 @@
+"use strict";
+
+var MessageSection = require('./MessageSection.react');
+var React = require('react');
+var ThreadSection = require('./ThreadSection.react');
+
+var ChatApp = React.createClass({
+
+    componentWillMount: function () {
+        this.props.router
+            .getModelObservable(this.props.modelId)
+            .observe(function (model) {
+                this.setState({ model: model });
```

Comment:
For larger apps this can't be too efficient. I'm not a React expert however presumably there is a bunch of diffing that goes on when this happens. The model in this app is quite small so it's probably fine. 

I noticed in the FB Flux example each component subscribes to it's store and sets it's own state. It's a more efficient model as the stores specifically emit changes and only the specific components will get the update. The same thing could be achieved in ESP by modeling the type of change and having the component subscribe to the model, filter on change type then select the sub graph they care about then apply that to their state. For example each component could have something similar to this:

```
    componentWillMount: function () {
        var _this this;
        this.props.router
            .getModelObservable(this.props.modelId)
           .where(function(m, e, c) { return m.SubPart.hasChanges; })
            .observe(function (model) {
                _this .setState(model.SubPart);
            });
    },
```

Then in the EventProcessor for `SubPart` you could set `hasChanges` and reset it in a `preEventProcessor`.